Creating 2d movement

Noel-Hann opened the pull request:
I have made the player object move in up, down, left, and right directions using the WASD keys. It also is rotated in the direction it is moving. I have additionally added a circle object to the player object to show the front, and a wall to test collisions

Noel-Hann committed:
Created 2d movement and rotation based on the WASD keys

The player object now moves in 4 directions following the WASD keys. The player also rotates to face the direction it is moving in. Additionally, I have added a circle to the front of the game object so that you can see which part of it is the front, as well as a wall to test that collision is working

Noel-Hann committed:
Update default-2022.dwlt

BenMeie committed:
Update from main

BenMeie committed:
Remove user settings

BenMeie reviewed (changes requested):
I think movement would feel better if when you pressed a second key while already moving in a direction it moved in that direction instead of just continuing in the existing direction.
Example:
1. Player presses W and moves up
2. While still holding W player presses D
3. Player then moves right
4. Player releases D
5. Player moves up

It would be great if the player moved a good bit faster. I don't want the game to feel super sluggish

As described in the issue, the 'front' object of the player should always follow the mouse, not be based on the direction of the player movement

Noel-Hann committed:
Changed movement to be less linear, and it now works smoothly with multiple inputs

Now if you press two buttons at the same time, the character will move diagonally, instead of just moving in the first direction. I still have to fix the player rotation to match its direction

Noel-Hann committed:
Updated player rotation to match player movement

Player rotation now fits with player movement. Player movement also now works in 8 directions, smoothly moving between them.

BenMeie committed:
Comment out debug log

BenMeie committed:
Merge branch 'main' into Creating_2d_movement
